File: 3. Longest Substring Without Repeating Characters.c
int lengthOfLongestSubstring(char* s) {
    if(s[0] == '\0') return 0;
    
    int maxLen = 0;
    int temp = 0;
    
    int i = 1;
    for(i; s[i] != '\0'; i++){
        for(int j=temp; j<i; j++){
            if(s[i] == s[j]){
                if(i - temp > maxLen){
                    maxLen = i - temp;
                }
                temp = j+1;
                i = temp;
            }
        }
    }
    if(i - temp > maxLen){
        maxLen = i - temp;
    }
    return maxLen;
}

File: 3-LongestSubstringWithoutRepeatingCharacters/3-LongestSubstringWithoutRepeatingCharacters.js
// Last updated: 6/22/2026, 12:55:30 PM
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function (s) {
    let subStr = [];
    let temp = [];
    let p = 0;
    for (let i = 0; i < s.length; i++) {
        if (!temp.includes(s[i])) {
            temp.push(s[i]);
        } else {
            subStr.push(temp.length);
            temp = [];
            i = p;
            p++;
        }
    }
    subStr.push(temp.length);
    subStr.sort((a, b) => b - a);
    return subStr[0]
};